### PR TITLE
[TOPSAIL] Add macOS support for topsail container

### PR DESCRIPTION
**Pull request by albertoperdomo2**
In order to be able to work with the Topsail container in MacOS, the container scripts needed a few adjustments that are effectively introduced by this PR. 

It mimics the `toolbx` behavior (simplistically) by using `podman` to create and enter Topsail environments.

### CODE REVIEW

File: launcher/topsail_build [hunk 1]
```diff
@@ -22,6 +22,13 @@ CMD []
...
 "
-podman build --tag $CONTAINER_IMAGE --file <(echo "$OVERLAY") --from $CONTAINER_IMAGE
+
+if [[ "$(uname)" == "Darwin" ]]; then
+ podman build --tag $CONTAINER_IMAGE --from $CONTAINER_IMAGE --file - <<EOF
+$OVERLAY
+EOF
+else
+ podman build --tag $CONTAINER_IMAGE --file <(echo "$OVERLAY") --from $CONTAINER_IMAGE
+fi
```
Comment (kpouget) [inline launcher/topsail_build:32]: can you keep only the mac version?

File: launcher/recreate [hunk 2]
```diff
@@ -21,7 +24,16 @@ fi
 
 if podman inspect --type container $CONTAINER_NAME &>/dev/null; then
   podman stop $CONTAINER_NAME || podman stop $CONTAINER_NAME
-  toolbox rm $CONTAINER_NAME
+
+  if [[ "$(uname)" == "Darwin" ]]; then
+   podman rm $CONTAINER_NAME
+  else
+   toolbox rm $CONTAINER_NAME
+  fi
```
Comment (kpouget) [inline launcher/recreate:32]: can you keep only the mac version?

File: launcher/recreate [hunk 3]
```diff
@@ -21,7 +24,16 @@ fi
...
 fi
 
-toolbox create $CONTAINER_NAME --image $CONTAINER_IMAGE
+if [[ "$(uname)" == "Darwin" ]]; then
```
Comment (kpouget) [inline launcher/recreate:35]: could you integrate something like this, in a common file, and use `if [[ "$HAS_TOOLBOX" == y ]]; then`
```
if toolbox --help > /dev/null 2>/dev/null;
then
  HAS_TOOLBOX=y
else
  HAS_TOOLBOX=y
fi
```

**Comment by kpouget** [comment]
@albertoperdomo2 , I've changed a bit the way we detect if `has_toolbox`, and I reshaped a bit the scripts to make them more reliable and readable
you'll let me know if you're fine with this version

**Comment by albertoperdomo2** [comment]
@kpouget looks great 👍🏽

**Comment by albertoperdomo2** [comment]
@kpouget can this be merged?

**Comment by kpouget** [comment]
yes, thanks @albertoperdomo2 , mergin!

**Pull request merged**